### PR TITLE
Implement view and delete functionality to buttons in admin/users

### LINKED ISSUE #799: Implement API functionality to buttons in admin/users
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Implements view and delete functionality to buttons in admin/users
- Adds dynamic user route in admin/users
- Moves update action from profile controller to form for reuse

Fixes #799

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

File: app/components/events/events-table.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ export default UiTable.extend({
     },
     openDeleteEventModal(id, name) {
       this.sendAction('openDeleteEventModal', id, name);
+    },
+    moveToUserDetails(id) {
+      this.sendAction('moveToUserDetails', id);
```
Comment (niranjan94) [inline app/components/events/events-table.js]: Please use the passed in actions directly for all of this.

**Comment by CosmicCoder96** [comment]
@geekyd @niranjan94 Done the changes, please review !

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@geekyd @niranjan94 Please review

File: app/controllers/admin/users/list.js [hunk 2]
```diff
@@ -57,10 +58,28 @@ export default Controller.extend({
       disableFiltering : true
     },
     {
-      template         : 'components/ui-table/cell/cell-buttons',
+      template         : 'components/ui-table/cell/admin/users/cell-actions',
       title            : '',
```
Comment (sumedh123) [inline app/controllers/admin/users/list.js]: Please include a 'Action' title here.

File: app/components/forms/user-profile-form.js [hunk 3]
```diff
@@ -38,7 +38,18 @@ export default Component.extend(FormMixin, {
     submit() {
       this.onValid(() => {
         this.set('isLoading', true);
-        this.sendAction('submit');
+        let user = this.get('user');
+        user.save()
```
Comment (sumedh123) [inline app/components/forms/user-profile-form.js]: We can do it in single line rather than having a variable user.

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-834-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@niranjan94 please review

**Comment by CosmicCoder96** [comment]
@niranjan94 Please review !

**Pull request merged**